ui/packages/consul-ui/app/services/client/http.js

```diff
@@ -72,21 +73,34 @@ const parseBody = function(strs, ...values) {
 
 const CLIENT_HEADERS = [CACHE_CONTROL, 'X-Request-ID', 'X-Range', 'Refresh'];
 export default class HttpService extends Service {
-  @service('dom')
-  dom;
-
-  @service('client/connections')
-  connections;
-
-  @service('client/transports/xhr')
-  transport;
-
-  @service('settings')
-  settings;
+  @service('dom') dom;
+  @service('env') env;
+  @service('client/connections') connections;
+  @service('client/transports/xhr') transport;
+  @service('settings') settings;
 
   init() {
     super.init(...arguments);
     this._listeners = this.dom.listeners();
+    this.parseURL = createURL(encodeURIComponent, obj => QueryParams.stringify(this.sanitize(obj)));
+  }
+
+  sanitize(obj) {
+    if (!this.env.var('CONSUL_NSPACES_ENABLED')) {
+      delete obj.ns;
+    } else {
+      if (typeof obj.ns === 'undefined' || obj.ns === null || obj.ns === '') {
+        delete obj.ns;
+      }
+    }
+    if (!this.env.var('CONSUL_PARTITIONS_ENABLED')) {
+      delete obj.partition;
+    } else {
+      if (typeof obj.partition === 'undefined' || obj.partition === null || obj.partition === '') {
+        delete obj.partition;
+      }
+    }
+    return obj;
```

Comment:
This is the stuff that removes `?ns` and `?partition` if you don't have them enabled. When I did this I remember thinking it's a shame that its a bit `mutate-y` but 🤷 for now at least.